src/JsonSchema/Constraints/ObjectConstraint.php

```diff
@@ -170,6 +175,26 @@ protected function &getProperty(&$element, $property, $fallback = null)
     }
 
     /**
+     * set and return a property on an object or array
+     *
+     * @param mixed $element Element to manipulate
+     * @param string $property  Property to set
+     * @param mixed $value Value to set
+     */
+    protected function &setProperty(&$element, $property, $value)
```

Comment:
Also--and I may be able to come up with something more helpful tomorrow--I remember writing a block of similar object-vs-array set code when I was doing type coercion, but I eventually managed to find a point in the code where the assignment could be done a little more naturally in the inner workings, and it wasn't necessary to shoehorn it in like this. I'm off to bed, but I'll see if I can remember what I'm talking about tomorrow.